### ISSUE TITLE
Move impala imports out of global namespace into tests where its used

### ISSUE THREAD

**Issue by matthewmturner** [ibis-project/ibis#2467]
> Should `IbisTestEnv` be using impala when its doing file system testing?

In Ibis `filesystems` means actually `hdfs`, and `hdfs` is only supported with Impala afaik. I thought on moving it inside the backend, but it's part of the public API, so users may be using it, so it requires more thought, discussion, and probably a deprecation warning first.

In any case, we should be calling pytest with `not hdfs`, and that should prevent `test_filesystems` from running. But I guess that import is in the global namespace, and that's causing the issue. Not sure why this is just failing in this PR. If you want to move that import inside the test/tests where it's being used (assuming it's not too many) that would be good. Probably better in a separate PR.

Thanks for all the work with this @matthewmturner, really happy to have you fixing all this technical debt.

_Originally posted by @datapythonista in https://github.com/ibis-project/ibis/pull/2435#issuecomment-707612410_

**Comment by matthewmturner** [comment]
@datapythonista 

`IbisTestEnv` is currently used as below:

```
ENV = IbisTestEnv()

class TestHDFSE2E(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        cls.ENV = ENV
        cls.tmp_dir = pjoin(cls.ENV.tmp_dir, util.guid())
        if cls.ENV.auth_mechanism in ['GSSAPI', 'LDAP']:
            print("Warning: ignoring invalid Certificate Authority errors")
        cls.hdfs = ibis.hdfs_connect(
            host=cls.ENV.nn_host,
            port=cls.ENV.webhdfs_port,
            auth_mechanism=cls.ENV.auth_mechanism,
            verify=(cls.ENV.auth_mechanism not in ['GSSAPI', 'LDAP']),
            user=cls.ENV.webhdfs_user,
        )
        cls.hdfs.mkdir(cls.tmp_dir)

@pytest.mark.superuser
class TestSuperUserHDFSE2E(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        cls.ENV = ENV
        cls.tmp_dir = pjoin(cls.ENV.tmp_dir, util.guid())
        if cls.ENV.auth_mechanism in ['GSSAPI', 'LDAP']:
            print("Warning: ignoring invalid Certificate Authority errors")
        # NOTE: specifying superuser as set in IbisTestEnv
        cls.hdfs = ibis.hdfs_connect(
            host=cls.ENV.nn_host,
            port=cls.ENV.webhdfs_port,
            auth_mechanism=cls.ENV.auth_mechanism,
            verify=(cls.ENV.auth_mechanism not in ['GSSAPI', 'LDAP']),
            user=cls.ENV.hdfs_superuser,
        )
        cls.hdfs.mkdir(cls.tmp_dir)
```

Would the idea be to move the global `ENV = IbisTestEnv()` and `from ibis.impala.tests.conftest import IbisTestEnv` into an `__init__` method for each of those classes with a variable like `self.ENV = IbisTestEnv()` and then update references to `cls.ENV` to `self.ENV`?

Thanks!

**Comment by datapythonista** [comment]
> Would the idea be to move the global `ENV = IbisTestEnv()` and `from ibis.impala.tests.conftest import IbisTestEnv` into an `__init__` method for each of those classes with a variable like `self.ENV = IbisTestEnv()` and then update references to `cls.ENV` to `self.ENV`?

I think should be as easy as replacing this:
```python
cls.ENV = ENV
```
By
```python
from ibis.impala.tests.conftest import IbisTestEnv
cls.ENV = IbisTestEnv()
```

And then removing the import and the initialization of `IbisTestEnv` from the global namespace.

**Comment by matthewmturner** [comment]
That makes sense and is much simpler...thx

I created #2470

**Comment by matthewmturner** [comment]
closed via #2470
